SensusService/Protocol.cs

```diff
@@ -1080,7 +1241,17 @@ public void StartWithUserAgreementAsync(string startupMessage, Action callback =
 
                         int consentCodeInt;
                         if (int.TryParse(consentCodeStr, out consentCodeInt) && consentCodeInt == consentCode)
-                            Start();
+                        {
+                            // start protocol or schedule start
+                            if (_startImmediately || (_startTime.Date.Equals(DateTime.Now.Date) && DateTime.Now.TimeOfDay > _startTime.TimeOfDay))
```

Comment:
If `DateTime.Now` is after `_startTime`, the protocol should be started immediately.